test/numbers.jl

```diff
@@ -2771,7 +2771,8 @@ let types = (Base.BitInteger_types..., BigInt, Bool,
     for S in types
         for op in (+, -)
             if S === Float16 # type instability here?
-                @test_broken @inferred Base.promote_op(op, S)
+                # broken, fixme then remove this branch
```

Comment:
I think the whole point of `@test_broken` is that it can show up in the report as broken. Since it doesn't work for this, it's better to just disable the test (i.e. remove the `@inferred`) and leave a comment. `@test_throws` is almost certainly the wrong thing to do.